src/corehost/cli/args.cpp

```diff
@@ -53,6 +53,16 @@ void setup_shared_store_paths(const hostpolicy_init_t& init, const pal::string_t
         args->dotnet_shared_store = own_dir;
         append_path(&args->dotnet_shared_store, RUNTIME_STORE_DIRECTORY_NAME);
         append_path(&args->dotnet_shared_store, init.tfm.c_str());
+
+        //Also add an architecture specific path to allow for unzipped developer cli experience
+        //where developer points runtime store operation output to "dotnet.exe" relative shared store folder
+
+        pal::string_t local_developer_store = own_dir ;
+        append_path(&local_developer_store, RUNTIME_STORE_DIRECTORY_NAME);
+        append_path(&local_developer_store, get_arch());
+        append_path(&local_developer_store, init.tfm.c_str());
+        local_developer_store.push_back(PATH_SEPARATOR);
+        (args->env_shared_store).push_back(local_developer_store);
```

Comment:
My thoughts are:

1. We shouldn't have the non-architecture locations anymore.  We should just always append the architecture.  Like we talked about in https://github.com/dotnet/sdk/issues/928#issuecomment-289089579.
2. We need to make the change for the global location as well, since it uses a non-architecture location.  Again, don't have the non-architecture locations at all.  It simplifies everything.